### PR TITLE
Removing imagemagick and using stb_image instead

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

This PR removes the optional ImageMagick dependency and adds image reader/writer using the stb_image files (included in DGtal).

The image import/export features in jpg, png, bmp and tga are now built-in in DGtal. 

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug mode.
- [x] All continuous integration tests pass (Github Actions & appveyor)

### CODE REVIEW

**Comment by dcoeurjo** [comment]
ready for review;)

File: src/DGtal/io/writers/STBWriter.ih [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file
```
Comment (kerautret) [inline src/DGtal/io/writers/STBWriter.ih:18]:
```suggestion
 * @file STBWriter.ih
```

**Review by kerautret — CHANGES_REQUESTED** [review]
Nice PR! Thanks, just some small suggest and a PR on your branch from another Magick ref to remove

File: src/DGtal/io/readers/GenericReader.h [hunk 2]
```diff
@@ -567,15 +558,10 @@ namespace DGtal
             typename TContainer::Point const pt (x,y);
             return RawReader< TContainer, TFunctor >::template importRaw<DGtal::Color>( filename, pt, aFunctor);
           }
-        else if ( extension == "gif" || extension == "jpg" || extension == "png" || extension == "jpeg" || extension == "bmp" )
+        else if ( extension == "tga" || extension == "jpg" || extension == "png" || extension == "jpeg" || extension == "bmp" )
```
[the remark this answers is not in the document]
Comment (dcoeurjo) [reply src/DGtal/io/readers/GenericReader.h:561]: no you're right. I'd need to add the methods in the Writer class.

File: .github/workflows/checkDoxygenDocumentation.sh [hunk 3]
```diff
@@ -54,7 +54,7 @@ fi
 
 
 ## We check src code consistency
-cd "$SRC_DIR/src"
+cd "$SRC_DIR/src/DGtal"
```
[the remark this answers is not in the document]
Comment (dcoeurjo) [reply .github/workflows/checkDoxygenDocumentation.sh:57]: yep, thx

**Comment by kerautret** [comment]
@dcoeurjo looks fine, and for the gif format ?

**Comment by dcoeurjo** [comment]
The method is still missing. I'll do it asap. Thx for the review

**Comment by dcoeurjo** [comment]
Ok gif added in the reader. Thx @kerautret

**Review by kerautret — APPROVED** [review]

**Comment by dcoeurjo** [comment]
Thx

**Pull request merged**